src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -327,7 +385,7 @@ private void ReadFrame()
                 indices = Buffer<byte>.CreateClean(imageDescriptor.Width * imageDescriptor.Height);
 
                 this.ReadFrameIndices(imageDescriptor, indices);
-                this.ReadFrameColors(indices, localColorTable ?? this.globalColorTable, imageDescriptor);
+                this.ReadFrameColors(ref image, ref previousFrame, indices, localColorTable ?? this.globalColorTable, imageDescriptor);
```

Comment:
`Image<T>` and `ImageFrame<T>` are already reference types?

Reply:
I had to remove `TPixel` type parameter from this class and move it `Decode` method. So `Image<TPixel> image` and `ImageFrame<TPixel> previousFrame` fields also removed and became local variables in `Decode<TPixel>` method. They marked with `ref` because they are reassigning down the call stack.